### DISCUSSION TITLE
Record-only and upload offsite for processing

### DISCUSSION THREAD

**Discussion by subzero** [mcguirepr89/BirdNET-Pi#394] [Q&A]
Hello!

I've been tinkering around with this and absolutely love the idea of it.
My Pi4 can handle it, but my Pi3 can only handle the workload as long as I actively cool it.

So my question is, is there a distributed setup I can grab a hold of, and have my sound-files dumped offsite for processing?
The setup seems simple enough as the analyzers already come as is to handle recorded files, but I feel like I'd waste 6 months just trying to get something semi-working, having a Pi record and offload the recordings.

If nothing pops up, I may look into doing something. It would enable older hardware like Pi 1/2's to be used for that purpose as well.
I'd like to avoid rolling Birdnet-Pi on and disabled all the services except recording!
I might have missed a project doing this already, which just means my google-fu is lacking!

Pi Recording -> Recordings are dumped offsite -> docker workers will handle it and have the coordinates of the mics set on them.

**Comment by ehpersonal38** [comment]
Hey there, glad you're enjoying the project so far - 

I think this is fairly easy to achieve how you describe. You'd need to harness the `scp` command to move all .wav files from Pi 1's `$HOME/BirdSongs/*` to the `$HOME/BirdSongs` directory on Pi 2. Then Pi 2 should pick these up and analyze them as normal, where the detections will get logged in birds.db.

An alternative and native way to do this would be to have Pi 1 keep it's "Live Audio Stream" service on, which will make its live audio stream available at `http://{BIRDNETPI_URL}/stream`. Then Pi 2 can use this link for its "RTSP Stream" value, which will start automatically pulling the data from that stream, analyzing & processing it, and adding the result to its birds.db. The downside is that this only supports 1 Pi recording at a time.

Let me know if you have any questions 🐤
